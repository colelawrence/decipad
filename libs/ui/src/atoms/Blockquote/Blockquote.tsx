/* eslint decipad/css-prop-named-variable: 0 */
import { useIsBlockActive } from '@decipad/react-contexts';
import { css } from '@emotion/react';
import { ReactNode } from 'react';
import { cssVar, setCssVar } from '../../primitives';
import { blockAlignment } from '../../styles';

const verticalPadding = '6px';

const styles = css(
  blockAlignment.blockquote.typography,
  setCssVar('currentTextColor', cssVar('weakTextColor')),
  {
    padding: `${verticalPadding} 30px ${verticalPadding} 12px`,
    borderLeft: `4px solid ${cssVar('highlightColor')}`,
    wordBreak: 'break-word',
  }
);

const activeStyles = css({
  backgroundColor: cssVar('highlightColor'),
  boxShadow: `0px 0px 0px 100vmin ${cssVar('highlightColor')}`,
  clipPath: `inset(0 -8px 0 -8px round 8px)`,
});

interface BlockquoteProps {
  readonly children: ReactNode;
}

export const Blockquote = ({
  children,
}: BlockquoteProps): ReturnType<React.FC> => {
  const isBlockActive = useIsBlockActive();
  return (
    <blockquote
      aria-label="column-content"
      css={[styles, isBlockActive && activeStyles]}
    >
      {children}
    </blockquote>
  );
};
