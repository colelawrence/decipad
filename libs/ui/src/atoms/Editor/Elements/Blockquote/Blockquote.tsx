import { css } from '@emotion/react';
import { PlatePluginComponent } from '@udecode/plate';
import {
  blockquote,
  cssVar,
  p16Regular,
  setCssVar,
} from '../../../../primitives';

const styles = css(p16Regular, {
  ...setCssVar('currentTextColor', cssVar('weakTextColor')),
  borderLeft: `5px solid ${blockquote.rgb}`,
  padding: '8px 24px',
  margin: '16px 0',
});

export const BlockquoteElement: PlatePluginComponent = ({
  attributes,
  children,
}) => {
  return (
    <blockquote css={styles} {...attributes}>
      {children}
    </blockquote>
  );
};
