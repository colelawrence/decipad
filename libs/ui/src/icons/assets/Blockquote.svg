<svg width="16" height="16" viewBox="0 0 16 16" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M7.49984 7.50008C7.49984 8.23648 6.9029 8.83341 6.1665 8.83341H4.49984C3.76346 8.83341 3.1665 8.23648 3.1665 7.50008V6.04241C3.1665 5.48421 3.39978 4.95141 3.80994 4.5728L5.33317 3.16675V5.16675H6.1665C6.9029 5.16675 7.49984 5.7637 7.49984 6.50008V7.50008Z" stroke="black" stroke-linecap="round" stroke-linejoin="round"/>
<path d="M12.8333 11.5001C12.8333 12.2365 12.2364 12.8334 11.5 12.8334H9.83333C9.09693 12.8334 8.5 12.2365 8.5 11.5001V10.0424C8.5 9.48421 8.73327 8.95141 9.14347 8.57281L10.6667 7.16675V9.16675H11.5C12.2364 9.16675 12.8333 9.76368 12.8333 10.5001V11.5001Z" fill="black" fill-opacity="0.1" stroke="black" stroke-linecap="round" stroke-linejoin="round"/>
</svg>
