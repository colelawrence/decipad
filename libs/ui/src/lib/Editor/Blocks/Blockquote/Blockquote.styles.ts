import { Text } from '@chakra-ui/layout';
import { chakra } from '@chakra-ui/system';

export const BlockquoteStyles = chakra(Text, {
  baseStyle: {
    pos: 'relative',
    p: '20px 30px',
    display: 'block',
    fontWeight: 'bold',
    color: 'green.300',
    _before: {
      content: `" "`,
      pos: 'absolute',
      left: 0,
      bottom: 0,
      w: '5px',
      h: '100%',
      bg: 'green.300',
      borderRadius: 'full',
    },
    my: 3,
  },
});
