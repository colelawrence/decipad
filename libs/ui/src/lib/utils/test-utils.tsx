import { ChakraProvider } from '@chakra-ui/react';
import { render, RenderOptions } from '@testing-library/react';
import React from 'react';
import { theme } from '../theme';

const AllTheProviders: React.FC = ({ children }) => {
  return (
    <ChakraProvider resetCSS theme={theme}>
      {children}
    </ChakraProvider>
  );
};

const customRender = (
  ui: JSX.Element,
  options?: Omit<RenderOptions, 'queries'>
) => render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from '@testing-library/react';
// override render method
export { customRender as render };
