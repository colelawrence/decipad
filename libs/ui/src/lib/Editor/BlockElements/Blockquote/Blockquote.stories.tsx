import React from 'react';
import { Blockquote } from './Blockquote.component';

export default {
  title: 'Editor/Block Elements/Blockquote',
};

export const B = () => (
  <Blockquote
    element={null}
    attributes={{
      'data-slate-node': 'element',
      ref: null,
    }}
  >
    The quick brown fox jumps over the lazy dog
  </Blockquote>
);

B.storyName = 'Blockquote';
