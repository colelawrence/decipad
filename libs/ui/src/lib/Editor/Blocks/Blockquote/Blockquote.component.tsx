import { useColorModeValue } from '@chakra-ui/react';
import React from 'react';
import { RenderElementProps } from 'slate-react';
import { BlockquoteStyles } from './Blockquote.styles';

interface BlockquoteProps extends RenderElementProps {
  children: string;
}

export const Blockquote = ({
  attributes,
  children,
}: BlockquoteProps): JSX.Element => {
  const bg = useColorModeValue('green.50', 'green.800');
  return (
    <BlockquoteStyles as="blockquote" bg={bg} {...attributes}>
      {children}
    </BlockquoteStyles>
  );
};
