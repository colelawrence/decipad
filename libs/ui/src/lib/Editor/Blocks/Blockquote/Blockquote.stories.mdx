import { Meta, Story, Canvas, Props } from '@storybook/addon-docs/blocks';
import { Blockquote } from './Blockquote.component';

<Meta
  title="Editor/Blocks/Blockquote"
  parameters={{
    docs: {
      source: {
        code: `<Blockquote
  element={{
    children: [{ text: 'The quick brown fox jumps over the lazy dog.' }],
  }}
  attributes={{ 'data-slate-node': 'element', ref: null }}
>
  The blockquote's text goes here
</Blockquote>`,
      },
    },
  }}
  argTypes={{
    children: {
      defaultValue: 'The quick brown fox jumps over the lazy dog.',
      description: 'The text of the component',
      type: 'string',
    },
  }}
  component={Blockquote}
/>

# Blockquote

The blockquote component is a fundamental block in our editor. It serves the purpose of displaying text that needs the reader's attention and focus.
The component is a slate element component meaning that it requires some slate specific props to be passed to the component when it is being rendered.
This component will not be used outside of Slate.

export const Template = (args) => <Blockquote {...args} />;

<Canvas>
  <Story name="Blockquote">{Template.bind({})}</Story>
</Canvas>

<Props story="Blockquote" />
