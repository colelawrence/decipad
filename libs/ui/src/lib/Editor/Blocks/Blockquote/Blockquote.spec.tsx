import { render } from 'test-utils';
import { Blockquote } from './Blockquote.component';

describe('Blockquote component', () => {
  it('renders', () => {
    const { container } = render(
      <Blockquote
        element={{
          children: [{ text: 'The quick brown fox jumps over the lazy dog.' }],
        }}
        attributes={{ 'data-slate-node': 'element', ref: null }}
      >
        The quick brown fox jumps over the lazy dog.
      </Blockquote>
    );

    expect(container).toBeDefined();
  });

  it('matches snapshot', () => {
    const { container } = render(
      <Blockquote
        element={{
          children: [{ text: 'The quick brown fox jumps over the lazy dog.' }],
        }}
        attributes={{ 'data-slate-node': 'element', ref: null }}
      >
        The quick brown fox jumps over the lazy dog.
      </Blockquote>
    );

    expect(container).toMatchSnapshot();
  });
});
