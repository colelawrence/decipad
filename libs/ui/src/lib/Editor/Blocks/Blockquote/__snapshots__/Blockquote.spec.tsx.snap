// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Blockquote component matches snapshot 1`] = `
.emotion-0 {
  position: relative;
  padding: 20px 30px;
  display: block;
  font-weight: var(--chakra-fontWeights-bold);
  color: var(--chakra-colors-green-300);
  margin-top: var(--chakra-space-3);
  margin-bottom: var(--chakra-space-3);
  background: var(--chakra-colors-green-50);
}

.emotion-0::before {
  content: " ";
  position: absolute;
  left: 0px;
  bottom: 0px;
  width: 5px;
  height: 100%;
  background: var(--chakra-colors-green-300);
  border-radius: var(--chakra-radii-full);
}

<div>
  <blockquote
    class="emotion-0"
    data-slate-node="element"
  >
    The quick brown fox jumps over the lazy dog.
  </blockquote>
</div>
`;
